### PR TITLE
Add stp root tests

### PR DESCRIPTION
**Pull request by StepanVovkPLV**
Depends on [bugfixes](https://github.com/PLVision/dent-testing/pull/161)

### CODE REVIEW

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/stp/test_stp_select_root.py [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+    # a priority value less than the default set
+    out = await Mstpctl.set(input_data=[{dent: [
+        {'parameter': 'treeportprio',
+         'bridge': bridge_names[0],
+         'port': list(loopback_ports.values())[1],
+         'mstid': 0,
+         'priority': 7}
+        ]}], parse_output=True)
+    assert out[0][dent]['rc'] == 0, 'Failed setting bridge priority'
+    # assert out[0][dent]['rc'] == 0, 'Failed to change priority of the bridge'  # RC = '-1', but command succeeds
```
Comment (SerhiyBoikoPLV) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/stp/test_stp_select_root.py]: uncomment or open an issue

**Review by yuriilisovskyi — APPROVED** [review]

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/stp/test_stp_ports.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    assert out[0][dent]['rc'] == 0, 'Failed to add bridge'
+
+    # 2. Enslave port to the bridge
+    out = await IpLink.set(input_data=[{dent: [
+        {'device': port,
+         'master': bridge} for port in ports]}])
+    assert out[0][dent]['rc'] == 0, 'Failed to set port state UP'
+
+    # 3. Change the MAC addresses the bridge
+    rc, _ = await dent_device.run_cmd(f'ifconfig {bridge} hw ether 22:BB:4D:85:E7:098')
```
Comment (SerhiyBoikoPLV) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/stp/test_stp_ports.py]: it is better to use `IpLink.set()` to change the mac addr

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/stp/test_stp_ports.py [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    # 7. Verify traffic is forwarded to port 1 and that port 2 doesn't receive any traffic
+    stats = await tgen_utils_get_traffic_stats(tgen_dev, 'Flow Statistics')
+    for row in stats.Rows:
+        if row['Traffic Item'] == traffic and row['Rx Port'] == tg_ports[1]:
+            err_msg = f'Expected 0.0 got : {float(row["Rx Rate (Mbps)"])}'
+            assert isclose(float(row['Rx Rate (Mbps)']), 0.0, abs_tol=tolerance), err_msg
+        if row['Traffic Item'] == traffic and row['Rx Port'] == tg_ports[0]:
+            err_msg = f'Expected 300 got : {float(row["Rx Rate (Mbps)"])}'
+            assert isclose(float(row['Rx Rate (Mbps)']), 300, rel_tol=tolerance), err_msg
+    await tgen_utils_stop_traffic(tgen_dev)
```
Comment (SerhiyBoikoPLV) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/stp/test_stp_ports.py]: redundant

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/stp/test_stp_select_root.py [hunk 4]
```diff
@@ -0,0 +1,640 @@
...
+    # 8. Verify the bridge is a root bridge
+    out = await Mstpctl.show(input_data=[{dent: [
+        {'parameter': 'bridge',
+         'bridge': bridge,
+         'options': '-f json'}]}], parse_output=True)
+    assert out[0][dent]['rc'] == 0, 'Failed to get bridge detail'
+    assert out[0][dent]['parsed_output'][0]['root-port'] == '', f'Bridge { bridge} is not a root bridge'
+
+
+@pytest.mark.parametrize('version', ['stp', 'rstp'])
```
Comment (SerhiyBoikoPLV) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/stp/test_stp_select_root.py]: You are using this in every test in this file. You can move this to `pytestmark` at L22 and remove this from all tests.

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/stp/test_stp_with_traffic.py [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+        for row in stats.Rows:
+            if row['Traffic Item'] not in [traffic, traffic_1] and row['Rx Port'] == tg_ports[1]:
+                err_msg = f'Expected 0.0 got : {float(row["Rx Rate (Mbps)"])}'
+                assert isclose(float(row['Rx Rate (Mbps)']), 0.0, abs_tol=tolerance), err_msg
+            if row['Traffic Item'] in [traffic, traffic_1] and row['Rx Port'] != tg_ports[1]:
+                err_msg = f'Expected 300 got : {float(row["Rx Rate (Mbps)"])}'
+                assert isclose(float(row['Rx Rate (Mbps)']), 300, rel_tol=tolerance), err_msg
+
+        # 8. Randomly select between STP/RSTP modes
+        version = choice(['stp', 'rstp'])
```
Comment (SerhiyBoikoPLV) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/stp/test_stp_with_traffic.py]: as discussed with @yuriilisovskyi - change the test steps 6-8 to be:
1) switch between STP-RSTP-STP
2) switch between RSTP-STP-RSTP

Choose either one or make the test parametrized and run both scenarios.

**Review by SerhiyBoikoPLV — CHANGES_REQUESTED** [review]

**Review by yuriilisovskyi — APPROVED** [review]

**Review by SerhiyBoikoPLV — APPROVED** [review]

**Pull request merged**